### PR TITLE
Cloudformation support for Events::Rule

### PR DESCRIPTION
**Pull request by gmcrocetti**
@bblommers , related with our ealier discussion in #2831.

I'm not quite sure about the cloudformation tests. Any tips to improve or create new ones ?
I'm also not quite sure about using `resource_name` during stack creation but I think it's enough for now.

### CODE REVIEW

File: tests/test_events/test_events.py [hunk 1]
```diff
@@ -1,15 +1,16 @@
-from moto.events.models import EventsBackend
-from moto.events import mock_events
 import json
 import random
 import unittest
 
 import boto3
+import sure  # noqa
```
Comment (gmcrocetti) [inline tests/test_events/test_events.py]: Tests were failing due to the lack of this import.

File: moto/events/models.py [hunk 2]
```diff
@@ -55,6 +55,24 @@ def remove_targets(self, ids):
             if index is not None:
                 self.targets.pop(index)
 
+    @classmethod
+    def create_from_cloudformation_json(
+        cls, resource_name, cloudformation_json, region_name
+    ):
+        properties = cloudformation_json["Properties"]
+        event_backend = events_backends[region_name]
+        event_name = properties.get("Name") or resource_name
```
Comment (bblommers) [inline moto/events/models.py]: Change this to ```properties.get("Name", resource_name)``` please - same in the delete. Slightly more readable IMO

Comment (gmcrocetti) [reply moto/events/models.py]: I used this "safer" approach to handle the case where `{"Name": ""}`. Is it possible ?

Comment (bblommers) [reply moto/events/models.py]: That is a very valid point! Didn't think about that possibility. Leave it as is then

**Comment by gmcrocetti** [comment]
> Hi @gmcrocetti, can you add two more tests?
> 
>     * Check what happens if you do not provide a rule name in the CF template
> 
>     * Check whether you can delete a rule via CF. So create a test that:
> 
> 
>     1. Creates a rule via CF
> 
>     2. Deletes a rule via CF
> 
>     3. Verify that no rules exist in the system

Added the required tests. Hope to hear from you soon. Thanks :)

**Pull request merged**